kernel/kernel-api/src/main/java/io/delta/kernel/exceptions/ConcurrentTransactionException.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.exceptions;
+
+import io.delta.kernel.annotation.Evolving;
+
+/**
+ * Thrown when concurrent transaction both attempt to update the same idempotent transaction.
```

Comment:
"Transaction" is pretty high-level here.

I'd also comment and point to some docs or the SetTransaction delta action to specifically indicate what we mean here, or mention the Transaction ID specified in the transaction builder

Else, to me, "Concurrent Transaction" seems like the same thing as "Concurrent Append"

Reply:
Added details and also link to the protocol txn identifiers section.